File: C3/Tarea17/M4/JS/ejercicio05.js
use ejercicio05;

db.createCollection('despachos');

db.despachos.insertMany([
    {
        "numero": 1,
        "capacidad": 5
    },
    {
        "numero": 2,
        "capacidad": 8
    },
    {
        "numero": 3,
        "capacidad": 10
    }
]);

db.createCollection('directores');

db.directores.insertMany([
    {
        "dni": "12345678A",
        "nombre": "Juan",
        "dni_jefe": null,
        "despacho": 1
    },
    {
        "dni": "23456789B",
        "nombre": "Maria",
        "dni_jefe": "12345678A",
        "despacho": 2
    },
    {
        "dni": "34567890C",
        "nombre": "Pedro",
        "dni_jefe": "12345678A",
        "despacho": 3
    },
    {
        "dni": "45678901D",
        "nombre": "Ana",
        "dni_jefe": "23456789B",
        "despacho": 1
    },
    {
        "dni": "56789012E",
        "nombre": "Laura",
        "dni_jefe": "23456789B",
        "despacho": 2
    },
    {
        "dni": "67890123F",
        "nombre": "Carlos",
        "dni_jefe": "34567890C",
        "despacho": 3
    }
]);

File: C2/Tarea18/proyecto_java/src/ejercicio05.java
import java.sql.Connection;

public class ejercicio05 {
    public static void main(String[] args) {
        Connection conec = null;
        String host = "127.0.0.1";
        String user = "root";
        String password = "";
        int port = 3306;
        String database = "";
        String table = "";
        String columasAfectadas = "";
        String values = "";

        // Hacer conexión con el localhost
        conec = BasesDeDatos.hacerConexionConBD(conec, host, port, database, user, password);
        // Crear base de datos para el ejercicio
        database = "despachos_directores_ej05";
        conec = BasesDeDatos.crearDataBase(conec, database);

        // Crear la primera tabla "despachos"
        table = "despachos";
        values = "(numero INT AUTO_INCREMENT PRIMARY KEY, capacidad INT)";
        conec = BasesDeDatos.crearTabla(conec, database, table, values);

        // Inserción de cinco registros en la primera tabla
        columasAfectadas = "(capacidad)";
        values = "(10), (10), (10), (10), (10)";
        conec = BasesDeDatos.insertarValues(conec, database, table, columasAfectadas, values);

        // Crear la segunda tabla "directores"
        table = "directores";
        values = "(dni VARCHAR(8) PRIMARY KEY, nombreAp NVARCHAR(100), dniJefe VARCHAR(8), despacho INT, FOREIGN KEY (dniJefe) REFERENCES directores (dni) ON DELETE CASCADE ON UPDATE CASCADE, FOREIGN KEY (despacho) REFERENCES despachos (numero) ON DELETE CASCADE ON UPDATE CASCADE)";
        conec = BasesDeDatos.crearTabla(conec, database, table, values);

        // Inserción de cinco registros en la segunda tabla
        columasAfectadas = "(dni, nombreAp, dniJefe, despacho)";
        values = "('11111111', 'Luis Aker', '11111111', 1), ('22222222', 'Juan Magan', '22222222', 2), ('33333333', 'Tony Flags', '33333333', 3), ('44444444', 'Frank Costello', '44444444', 4), ('55555555', 'Charles Luciano', '55555555', 5)";
        conec = BasesDeDatos.insertarValues(conec, database, table, columasAfectadas, values);
    }
}